frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1112,6 +1112,78 @@ pub mod pallet {
 			<QueuedSolution<T>>::put(solution);
 			Ok(())
 		}
+
+		/// Rotates the round which results in setting the current phase to
+		/// `Phase::Off` and removes all the current solutions.
+		///
+		/// The dispatch origin for this call must be `ForceOrigin`.
+		#[pallet::call_index(5)]
+		#[pallet::weight({
+			if current_phase.is_signed() {
+				T::WeightInfo::force_rotate_round_from_signed()
+			}else {
+				T::WeightInfo::force_rotate_round()
+			}
+		})]
+		pub fn force_rotate_round(
+			origin: OriginFor<T>,
+			current_phase: Phase<T::BlockNumber>,
+		) -> DispatchResult {
+			T::ForceOrigin::ensure_origin(origin)?;
+			ensure!(current_phase == Self::current_phase(), <Error<T>>::CallNotAllowed);
+
+			Self::do_force_rotate_round();
+
+			Ok(())
+		}
+
+		/// Initializes an open signed phase. If the current phase is not
+		/// `Phase::Off` proceeds to the next round.
+		///
+		/// The dispatch origin for this call must be `ForceOrigin`.
+		#[pallet::call_index(6)]
+		#[pallet::weight({
+			if current_phase.is_signed() {
+				T::WeightInfo::on_initialize_open_signed().saturating_add(T::WeightInfo::force_rotate_round_from_signed())
+			}else if !current_phase.is_off() {
+				T::WeightInfo::on_initialize_open_signed().saturating_add(T::WeightInfo::force_rotate_round())
+			}else {
+				T::WeightInfo::on_initialize_open_signed()
+			}
+		})]
+		pub fn force_start_signed_phase(
+			origin: OriginFor<T>,
+			current_phase: Phase<T::BlockNumber>,
+		) -> DispatchResult {
+			T::ForceOrigin::ensure_origin(origin.clone())?;
+			ensure!(current_phase == Self::current_phase(), <Error<T>>::CallNotAllowed);
+
+			// If the current phase is not `Phase::Off` proceed to the next round.
+			if !Self::current_phase().is_off() {
+				Self::do_force_rotate_round();
+			}
+
+			match Self::create_snapshot() {
```

Comment:
Yes, I agree, that makes more sense.

Reply:
The current code is kinda assuming that `T::WeightInfo::on_initialize_open_signed()` is the weight of the `create_snapshot`. 

Generally speaking, if the weight calculation is correct, if is fine to allow this to be an extrinsic. But, if the weight of this operation is more than what is allowed in a block, this transaction can never be called.

You can do a back of an envelope calculation and see if the current weight will fit in a typical 2s block and make a decision based on that.

Also, all of the transactions in this PR are good candidates for `DispatchClass::Operational`.

Reply:
> The current code is kinda assuming that `T::WeightInfo::on_initialize_open_signed()` is the weight of the `create_snapshot`.

Yes, it is assuming that, and AFAIU it is a bad assumption. `create_snapshot` accounts for its own weight by calling `register_weight` while the weight attribute of the extrinsic only mentions the weight consumption of `on_initialize_open_signed`. I should probably implement the above mention solution.

Reply:
@kianenigma @Ank4n I have implemented this solution:

> This could potentially be via a new StorageItem ForcePhase which could store an Option<Phase>. If ForcePhase is set to Some(Signed), the pallet tries to transition to Signed Phase in the next on_initialize and resets ForcePhase to None.